### PR TITLE
Rewrite document converter.

### LINKED ISSUE #9: Coverage data doesn't match report
Sorry for bothering you again. Now that the API plugin in the dev version is installed I do get a coverage report. However I can't really make sense of these numbers. I can't upload images to GitHub so text will have to do:

I get

LLVM Function

* Covered: 403
* Missed: 22

However looking at the totals section in coverage.json it has345 functions and 4 covered (which is expected because I'm only running a single test case).

I could understand if the files outside the repository are filtered, however there are more than the total number of functions which I don't get.

### PR DESCRIPTION
**Pull request by cizezsy**
Use data in summary section, remove function child summary chart.

Fix #9.

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/llvm/LLVMCovReportDocumentConverter.java [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+            JsonNode dataObj = dataArr.get(i);
+
+            processDataObj(dataObj, dataEle, document);
+        }
+
+        return document;
+    }
+
+    /**
+     * parse each data object in JSON, and convert it to data element and them to document.
```
Comment (jeffpearce) [inline src/main/java/io/jenkins/plugins/llvm/LLVMCovReportDocumentConverter.java]:
```suggestion
     * parse each data object in JSON, and convert it to data element and then to document.
```

File: src/main/java/io/jenkins/plugins/llvm/LLVMCovReportDocumentConverter.java [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+        if (!report.get("type").asText().equals("llvm.coverage.json.export")) {
+            throw new CoverageException("Unsupported Json file - type must be llvm.coverage.json.export");
+        }
+
+        Element reportEle = document.createElement("report");
+        reportEle.setAttribute("name", "llvm-cov");
+        document.appendChild(reportEle);
+
+        JsonNode dataArr = report.get("data");
```
Comment (jeffpearce) [inline src/main/java/io/jenkins/plugins/llvm/LLVMCovReportDocumentConverter.java:33]: Is it possible for the report to not have a data node? If so would it be good to verify and throw an explicit exception here?

File: src/main/java/io/jenkins/plugins/llvm/LLVMCovReportDocumentConverter.java [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    private List<Element> processFiles(JsonNode files, Document document) throws CoverageException {
+        List<Element> fileElements = new LinkedList<>();
+        for (int i = 0; i < files.size(); i++) {
+            JsonNode file = files.get(i);
+
+            Element fileEle = document.createElement("file");
+            fileEle.setAttribute("filename", file.get("filename").asText());
+
+            JsonNode segments = file.get("segments");
+
```
Comment (jeffpearce) [inline src/main/java/io/jenkins/plugins/llvm/LLVMCovReportDocumentConverter.java:110]:
```suggestion
```

File: src/main/java/io/jenkins/plugins/llvm/LLVMCovReportDocumentConverter.java [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+            } else {
+                return path.getParent();
+            }
+        })).forEach((parentPath, fileEles) -> {
+            Element directoryEle = document.createElement("directory");
+            directoryEle.setAttribute("name", parentPath);
+            fileEles.forEach(directoryEle::appendChild);
+            dataEle.appendChild(directoryEle);
+        });
+
```
Comment (jeffpearce) [inline src/main/java/io/jenkins/plugins/llvm/LLVMCovReportDocumentConverter.java:90]:
```suggestion
```

File: src/main/java/io/jenkins/plugins/llvm/LLVMCoverageParser.java [hunk 5]
```diff
@@ -45,17 +46,31 @@ protected CoverageResult processElement(Element current, CoverageResult parentRe
...
+                int hits = Integer.parseInt(hitsString);
+                int number = Integer.parseInt(lineNumber);
+
+                parentResult.paint(number, hits);
                 break;
             default:
                 break;
         }
+
+
```
Comment (jeffpearce) [inline src/main/java/io/jenkins/plugins/llvm/LLVMCoverageParser.java]:
```suggestion
```

**Comment by cizezsy** [comment]
@jeffpearce Fixed them, thanks~

**Pull request merged**